src/sagas/admin.js

```diff
@@ -68,33 +90,56 @@ export function* deployStore({ payload }) {
     });
     const gasPrice = (yield web3.eth.getGasPrice()) * 5;
     const gasLimit = (yield deployment.estimateGas()) * 2;
+    const transactionCostInEthers = web3.utils.fromWei(
+      (gasPrice * gasLimit).toString(),
+      "ether"
+    );
 
-    const txHash = yield sendTxWrapper({
-      txObject: deployment,
-      gasPrice,
-      gasLimit,
-      fromAddress
-    });
-
-    yield put({
-      type: types.DEPLOYING_STORE_TX_SUBMITTED,
-      payload: txHash
-    });
+    if (accountBalance >= transactionCostInEthers) {
+      let messageToast = null;
+      messageToast = toast("Deploying new certificate store...", {
+        autoClose: false
+      });
+      const txHash = yield sendTxWrapper({
+        txObject: deployment,
+        gasPrice,
+        gasLimit,
+        fromAddress
+      });
 
-    let txReceipt;
+      yield put({
+        type: types.DEPLOYING_STORE_TX_SUBMITTED,
+        payload: txHash
+      });
 
-    while (!txReceipt) {
-      yield take(applicationTypes.TRANSACTION_MINED);
-      txReceipt = yield select(getTransactionReceipt, txHash); // this returns undefined if the transaction mined doesn't match the txHash we're waiting for
-    }
+      let txReceipt;
 
-    yield put({
-      type: types.DEPLOYING_STORE_SUCCESS,
-      payload: {
-        contractAddress: txReceipt.contractAddress,
-        txHash: txReceipt.transactionHash
+      while (!txReceipt) {
+        yield take(applicationTypes.TRANSACTION_MINED);
+        txReceipt = yield select(getTransactionReceipt, txHash); // this returns undefined if the transaction mined doesn't match the txHash we're waiting for
       }
-    });
+
+      yield put({
+        type: types.DEPLOYING_STORE_SUCCESS,
+        payload: {
+          contractAddress: txReceipt.contractAddress,
+          txHash: txReceipt.transactionHash
+        }
+      });
+      toast.update(messageToast, {
+        render: "Successfully deployed a new certificate store.",
+        type: toast.TYPE.SUCCESS,
+        className: "rotateY animated",
+        autoClose: 2000
+      });
+    } else {
+      const errorMessage = "Insufficient Ethers in wallet.";
```

Comment:
The error may not necessary be because of insufficient ethers. If we cannot determine the error from the error message, send the user to etherscan or just put a more generic error message.

Reply:
Updated.